src/czt.jl

```diff
@@ -267,30 +269,38 @@ function czt_1d(xin, plan::CZTPlan_1D)
     # which (intentionally) leads to non-real results for even-sized arrays at non-unit zoom
 
     L = size(plan.fft_fv, plan.d)
-    nsz = ntuple((dd) -> (dd==plan.d) ? L : size(xin, dd), Val(ndims(xin))) 
+    nsz = ntuple((dd) -> (dd==plan.d) ? L : size(xin, dd), Val(D))  # sizes may vary per dimension, so a new array has to be generated
     # append zeros
     y = zeros(eltype(plan.aw), nsz)
```

Comment:
If you want this CUDA runable, do
```
y = similar(xin, eltype(plan.aw), nsz)
fill!(y, 0)
```

In my code, I try to avoid `zeros` or `ones`.